hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/NativeLibraryLoader.java

```diff
@@ -139,7 +139,7 @@ private Optional<File> copyResourceFromJarToTemp(final String libraryName)
       // create a temporary file to copy the library to
       final File temp = File.createTempFile(libraryName, getLibOsSuffix(),
           new File(Optional.ofNullable(System.getProperty(NATIVE_LIB_TMP_DIR))
-              .orElse("")));
+              .orElse("")).getAbsoluteFile());
```

Comment:
Was it causing problem without `getAbsoluteFile()`?

Reply:
yeah. When parent has path empty. It resorts to fs.defaultParent() which is "/" so the test case fails with no permission on "/" to create a temp file.
https://github.com/bpupadhyaya/openjdk-8/blob/45af329463a45955ea2759b89cb0ebfe40570c3f/jdk/src/share/classes/java/io/File.java#L363